### PR TITLE
Decode non-XML entities for translations

### LINKED ISSUE #3857: Sitemaps: remove illegal entities from sitemap and XSLT
XML only accepts a few entities, and will return errors when you use wrong entities:
- https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references#Predefined_entities_in_XML
- https://support.google.com/webmasters/answer/183668?hl=en&rd=1

However, in some translations of [this line](https://github.com/Automattic/jetpack/blob/master/modules/sitemaps/sitemap-xsl.php#L94), we use HTML entities like `&rsquo;`.

```
<p>Vous pouvez trouver plus d&rsquo;information sur les sitemaps XML sur <a href="http://sitemaps.org">sitemaps.org</a></p>
```

That creates an error and the sitemap can't be properly displayed.

While we can fix the issue by using `&apos;` instead of `&rsquo;` (I already changed the French translation to do that), it would be nice to implement a fail safe, to make sure no invalid entities are ever used.

Reported in 2667562-t

### PR DESCRIPTION
**Pull request by akirk**
Fixes #3857

#### Changes proposed in this Pull Request:
This replaces named HTML entities, for example `&rsquo;` as it is used in French translations, so that it doesn't break the XSLT.

To test:
 - set the language to French
 - delete the `jetpack_sitemap_xsl` transient
 - access `/sitemap.xsl` and check that no named entity appears there

### CODE REVIEW

File: modules/sitemaps/sitemap-xsl.php [hunk 1]
```diff
@@ -90,8 +100,8 @@
 			<body>
 				<div id="description">
 					<h1>' . esc_html__( 'XML Sitemap', 'jetpack' ) . '</h1>
-					<p>' . wp_kses( sprintf( __( 'This is an XML Sitemap generated by <a href="%s" target="_blank">Jetpack</a>, meant to be consumed by search engines like <a href="%s" target="_blank">Google</a> or <a href="%s" target="_blank">Bing</a>.', 'jetpack' ), 'http://jetpack.com/', 'https://www.google.com', 'https://www.bing.com/' ), array( 'a' => array( 'href' => true, 'title' => true ) ) ) . '</p>
-					<p>' . wp_kses( sprintf( __( 'You can find more information on XML sitemaps at <a href="%s" target="_blank">sitemaps.org</a>', 'jetpack' ), 'http://sitemaps.org' ), array( 'a' => array( 'href' => true, 'title' => true ) ) ) . '</p>
+					<p>' . wp_kses( sprintf( html_entity_decode_for_xml( __( 'This is an XML Sitemap generated by <a href="%s" target="_blank">Jetpack</a>, meant to be consumed by search engines like <a href="%s" target="_blank">Google</a> or <a href="%s" target="_blank">Bing</a>.', 'jetpack' ), 'http://jetpack.com/', 'https://www.google.com', 'https://www.bing.com/' ), array( 'a' => array( 'href' => true, 'title' => true ) ) ) ) . '</p>
```
Comment (kraftbj) [inline modules/sitemaps/sitemap-xsl.php]: What about using `ent2ncr` instead? It's what Core uses for RSS feeds to get valid XML. https://developer.wordpress.org/reference/functions/ent2ncr/

Comment (akirk) [inline modules/sitemaps/sitemap-xsl.php]: Perfect, didn't know about it :)

**Comment by tobiasehlert** [comment]
It looks to me that there is also an issue with the translation on lines 93 in Swedish.
`&Auml;ndra frekvens` should be `Ändra frekvens` instead I guess.

This needs to be fixed somehow
line 93 `<th>' . esc_html__( 'Change Frequency', 'jetpack' ) . '</th>`

**Comment by akirk** [comment]
@tobiasehlert Thanks for noticing, I have now covered all translated text with `ent2ncr()`.

**Comment by jeherve** [comment]
This seems to work well in my tests. LGTM!

**Pull request merged**